### PR TITLE
8328035: Convert javax/swing/text/html/TableView/7030332/bug7030332.java applet test to main

### PR DESCRIPTION
**Pull request by TejeshR13**
Convert javax/swing/text/html/TableView/7030332/bug7030332.java applet test to main based test using PassFailJFrame.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328035](https://bugs.openjdk.org/browse/JDK-8328035): Convert javax/swing/text/html/TableView/7030332/bug7030332.java applet test to main (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [8c40a81b](https://git.openjdk.org/jdk/pull/18260/files/8c40a81b6b110d34e860c1c35996a26e3cd80685)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18260/head:pull/18260` \
`$ git checkout pull/18260`

Update a local copy of the PR: \
`$ git checkout pull/18260` \
`$ git pull https://git.openjdk.org/jdk.git pull/18260/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18260`

View PR using the GUI difftool: \
`$ git pr show -t 18260`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18260.diff">https://git.openjdk.org/jdk/pull/18260.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18260#issuecomment-1993698205)

### CODE REVIEW

File: test/jdk/javax/swing/text/html/TableView/7030332/bug7030332.java [hunk 1]
```diff
@@ -21,54 +21,53 @@
...
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug7030332
+ */
+
+public class bug7030332 {
+    public static void main(String[] args) throws Exception {
+        String testInstructions = """
+                Compare Golden Images with rendered JEditorPane.
+                They should looks similar in each line.
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/TableView/7030332/bug7030332.java:47]:
```suggestion
                They should look similar in each line.
```

File: test/jdk/javax/swing/text/html/TableView/7030332/bug7030332.java [hunk 2]
```diff
@@ -21,54 +21,53 @@
...
+import javax.swing.SwingConstants;
 
-import javax.swing.*;
-import java.awt.*;
 import java.net.URL;
 
-public class bug7030332 extends JApplet {
+/* @test
+ * @bug 7030332
+ * @summary Default borders in tables looks incorrect JEditorPane
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/text/html/TableView/7030332/bug7030332.java:37]: Is something missing in summary?

Comment (TejeshR13) [reply test/jdk/javax/swing/text/html/TableView/7030332/bug7030332.java]: Modified it.

**Review by kumarabhi006 — APPROVED** [review]
LGTM.

**Comment by prsadhuk** [comment]
I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted
Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...

image

**Comment by TejeshR13** [comment]
> I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...
> 
> image

You mean to say the border of JEditorPane is 1 pixel thicker ?

**Comment by prsadhuk** [comment]
> > I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...
> > image
> 
> You mean to say the border of JEditorPane is 1 pixel thicker ?

it doesn't look exactly same subjectively...

image

**Comment by TejeshR13** [comment]
> > > I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...
> > > image
> > 
> > 
> > You mean to say the border of JEditorPane is 1 pixel thicker ?
> 
> it doesn't look exactly same subjectively...
> 
> image

Yeah, but in instructions they say it should look similar, so wasn't sure whether to problem list or proceed with conversion. You think we should problem list?

**Comment by prsadhuk** [comment]
> > > > I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...
> > > > image
> > > 
> > > 
> > > You mean to say the border of JEditorPane is 1 pixel thicker ?
> > 
> > 
> > it doesn't look exactly same subjectively...
> > image
> 
> Yeah, but in instructions they say it should look similar, so wasn't sure whether to problem list or proceed with conversion. You think we should problem list?

I am not sure...I think there was some fix regarding border stroke sometime back either by @honkar-jdk or @DamonGuy . Maybe expectation should be somewhat different now in which case, probably the instructions should be amended accordingly...Let them verify once..

**Comment by prsadhuk** [comment]
> > > > > I am not sure but to me the images dont seem to match in windows11..it seems JEditorPane image is 1 pixel thicker than golden images...in which case, it needs to be problemlisted Also, I guess this has potential of making it automated by loading the golden images and JEditorPane in BufferedImage and then compare...
> > > > > image
> > > > 
> > > > 
> > > > You mean to say the border of JEditorPane is 1 pixel thicker ?
> > > 
> > > 
> > > it doesn't look exactly same subjectively...
> > > image
> > 
> > 
> > Yeah, but in instructions they say it should look similar, so wasn't sure whether to problem list or proceed with conversion. You think we should problem list?
> 
> I am not sure...I think there was some fix regarding border stroke sometime back either by @honkar-jdk or @DamonGuy . Maybe expectation should be somewhat different now in which case, probably the instructions should be amended accordingly...Let them verify once..

or maybe run the test with uiScale 1.0 as I believe the golden images are generated without any scaling and windows has default 1.25/1.5 scaling

**Comment by TejeshR13** [comment]
With scaling 1.0 it'll appear this way.
image

**Comment by prsadhuk** [comment]
> @prsadhuk With scaling 1.0 it'll appear this way. I think it's supposed to be 1.0 scaling and its looking similar now. image

I am not sure..it still seems 1.25 bigger than golden image..."similar" word has different connotation and is very subjective....is it intentional that "same" was not mentioned?
What if we say, Earth and Mars are similar as they are planets...

**Comment by TejeshR13** [comment]
> > @prsadhuk With scaling 1.0 it'll appear this way. I think it's supposed to be 1.0 scaling and its looking similar now. image
> 
> I am not sure..it still seems 1.25 bigger than golden image..."similar" word has different connotation and is very subjective....is it intentional that "same" was not mentioned? What if we say, Earth and Mars are similar as they are planets...

Then shall I problemlist it?

**Comment by prsadhuk** [comment]
Ideally yes, but you can discuss in the meeting since harshita/damon has not come back on this..

**Comment by honkar-jdk** [comment]
> I am not sure...I think there was some fix regarding border stroke sometime back either by @honkar-jdk or @DamonGuy . Maybe expectation should be somewhat different now in which case, probably the instructions should be amended accordingly...Let them verify once..

@prsadhuk @TejeshR13  I do see slight difference is borders on fractional scales 1.25,1.75. This looks similar to the border scaling issue fixed as part https://github.com/openjdk/jdk/pull/11571, https://github.com/openjdk/jdk/pull/10274. The border scaling fix was extended to line, etched borders and borders for JInternalFrame. 

It is slightly different, in the sense html is parsed to paint the border of desired thickness but I believe the underlying issue is the same - painting of borders on fractional scales on Windows.

**Comment by prrace** [comment]
This test was written before there was hidpi. So I am sure it is 100% so definitely you should specify that in the test to have any chance.
The height of the text doesn't matter, nor spacing inside the cells.
I'd say it is "similar enough" with 1.0 scaling. There's not been a bug about this yet.

**Comment by TejeshR13** [comment]
> This test was written before there was hidpi. So I am sure it is 100% so definitely you should specify that in the test to have any chance. The height of the text doesn't matter, nor spacing inside the cells. I'd say it is "similar enough" with 1.0 scaling. There's not been a bug about this yet.

Updated the instructions and scaling set to 1.0.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**